### ISSUE TITLE
Improve date format selection UI

### ISSUE THREAD

**Issue by aaronrothschild** [mattermost/mattermost-plugin-agenda#66]
Currently, the user needs to type in a date format to be used - manually.  This introduces errors and confusion for the end-user.  It would be better if we just gave them 8 choices in a list and didn't have to type anything in. 

- (***) Improve the UI to use a radio or dropdown for the user to select their desired date format from a few options rather 
- Add validation that the hashtag before the date is a minimum of 3 characters to ensure meeting items create valid hashtags

**Comment by icy-meteor** [comment]
Yes, the current format is really painful and contributes to a bad UX. 
Do we have any design for the proposed change?

**Comment by icy-meteor** [comment]
Is this issue the same as https://github.com/mattermost/mattermost-plugin-agenda/issues/17? @hanzei

**Comment by hanzei** [comment]
As far as I understand https://github.com/mattermost/mattermost-plugin-agenda/issues/17 it's about validating the time format before saving it. That seems to be easier to accomplish then this ticket. Hence, I'm inclined to keep both open.

**Comment by aaronrothschild** [comment]
Discussion with some context about this issue: https://community.mattermost.com/core/pl/97x3akmh8jdc5mdg4afyedbhmr

**Comment by aaronrothschild** [comment]
Let's add these options:

"Month-day" (ie: Dec-4)
"day-Month" (ie: 4-Dec)
"month-day" (ie: 12-4 )
"day-month" (ie: 4-12 )
"year-month-day" (ie: 2020-12-4) 

We will need a second field called "hashtag prefix" which the user can specify the "text" part of the hastag (such as "incident-room" or "TeamA") which is then concatenated with the date format they've chosen from the dropdown.  

They would be rendered as:
"#TeamA-12-4" (assuming they chose the "month-day" option)

Somrthing like this:
![image](https://user-images.githubusercontent.com/915956/101166111-9ac3a880-35ec-11eb-81db-8f647cc88530.png)

**Comment by icy-meteor** [comment]
Got it, thanks! 
@hanzei Could you assign this issue to me? :)

**Comment by hanzei** [comment]
Sure. Thanks for working on this @icy-meteor :+1:
